doc/OVERVIEW.md

```diff
@@ -0,0 +1,470 @@
+# Introduction
```

Comment:
I haven't reviewed this in detail. Please just check that the embedded API snippets are still accurate.

Reply:
Ah, thanks for the diff link!